src/Shared/NewSwapModal.razor

```diff
@@ -0,0 +1,503 @@
+@using Grpc.Core
+@using NBXplorer.DerivationStrategy
+@using NBitcoin
+@using Blazorise
+<Modal @ref="_modalRef">
+    <ModalContent Centered>
+        <ModalHeader>
+            <ModalTitle>New swap</ModalTitle>
+            <CloseButton />
+        </ModalHeader>
+        <ModalBody Class="overflow-y-auto" Style="max-height: 70vh;">
+            <Fields Flex="Flex.Column">
+                <Field>
+                    <FieldLabel>Swap provider</FieldLabel>
+                    <div class="d-flex">
+                        <Select TValue="string" @bind-Value="_selectedProvider">
+                            @foreach (var provider in Enum.GetValues<SwapProvider>())
+                            {
+                                <SelectItem Value="@provider.ToString()">@provider.ToString()</SelectItem>
+                            }
+                        </Select>
+                    </div>
+                </Field>
+                <Field>
+                    <FieldLabel>Swap direction</FieldLabel>
+                    <div class="d-flex">
+                        <Select TValue="string" @bind-Value="_selectedDirection">
+                            @foreach (var direction in Enum.GetValues<SwapDirection>())
+                            {
+                                <SelectItem Value="@direction.ToString()">@direction.ToString()</SelectItem>
+                            }
+                        </Select>
+                    </div>
+                </Field>
+                <Field>
+                    <FieldLabel>Amount</FieldLabel>
+                    <div class="d-flex">
+                        <NumericPicker TValue="decimal"
+                                    @bind-Value="_amountBtc"
+                                    CurrencySymbolPlacement="CurrencySymbolPlacement.Suffix"
+                                    CurrencySymbol=" BTC"
+                                    Decimals="8"
+                                    Step="0.00001000m"
+                                    Min="0.00001000m">
+                        </NumericPicker>
+                    </div>
+                    <FieldHelp>
+                        $@Math.Round(PriceConversionService.BtcToUsdConversion(_amountBtc, _btcPrice), 2)
+                    </FieldHelp>
+                </Field>
+                @if (_selectedDirection == SwapDirection.Out)
+                {
+                    @RenderNodeSelector("Origin Node")
+                    @RenderWalletSelector("Destination Wallet")
+                }
+                else
+                {
+                    @RenderWalletSelector("Origin Wallet")
+                    @RenderNodeSelector("Destination Node")
+                }
+                <Field>
+                    <FieldLabel>Max Service Fees</FieldLabel>
```

Comment:
I dont think we need to set those here, rather I would only add as a % basis, max swap routing fee and then calculate, why? 90% of the cossts to loop are routing fees, services fees are whatever they tell us thoughs nice to have, its not a big deal

Reply:
refactored. Also check the prepayAmount, I'm also taking the % that we set up